.github/workflows/python.yml

```diff
@@ -141,5 +141,25 @@ jobs:
 
       - uses: actions/upload-artifact@v3
         with:
-          name: cibw-wheels-${{ matrix.os }}-${{ strategy.job-index }}
+          name: release-${{ matrix.os }}-${{ strategy.job-index }}
           path: ./wheelhouse/*.whl
+
+  upload_pypi:
+    name: Publish on PyPI
+    needs: [build_wheels, build]
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: release-*
+          merge-multiple: true
+          path: dist
+
+      - name: Upload Release Assets to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
@dbaston you'll need to create an account on pypi (if not made yet) and then create a token https://pypi.org/help/#apitoken (best practices) and then add it in the GHA Secrets (`PYPI_PASSWORD`)